src/prefecto/ext/snowflake.py

```diff
@@ -0,0 +1,620 @@
+            ```
+        """  # noqa
+        connect_params = {
+            # required to track task's usage in the Snowflake Partner Network Portal
+            "application": "Prefect_Snowflake_Collection",
+            **self.credentials.model_dump(
+                exclude_unset=True, exclude={"block_type_slug"}
+            ),
+            **connect_kwargs,
+        }
+
+        for key, value in connect_params.items():
+            if isinstance(value, (SecretStr, SecretBytes)):
+                connect_params[key] = connect_params[key].get_secret_value()
+
+        # set authenticator to the actual okta_endpoint
+        if connect_params.get("authenticator") == "okta_endpoint":
+            endpoint = connect_params.pop("endpoint", None) or connect_params.pop(
+                "okta_endpoint", None
+            )  # okta_endpoint is deprecated
+            connect_params["authenticator"] = endpoint
+
+        private_der_key = self.credentials.resolve_private_key()
+        if private_der_key is not None:
+            connect_params["private_key"] = private_der_key
+            connect_params.pop("password", None)
+            connect_params.pop("private_key_passphrase", None)
+
+        return PrefectoSnowflakeConnection(**connect_params)
+
+    def get_connection(self, **connect_kwargs: Any) -> PrefectoSnowflakeConnection:
+        """
+        Returns an authenticated connection that can be
+        used to command from Snowflake databases.
+
+        Args:
+            **connect_kwargs: Additional arguments to pass to
+                `snowflake.connector.connect`.
+
+        Returns:
+            The authenticated SnowflakeConnection.
+
+        Examples:
+            ```python
+            from prefecto.ext.snowflake import PrefectoSnowflakeCredentials as SnowflakeCredentials
+            from prefecto.ext.snowflake import PrefectoSnowflakeConnector as SnowflakeConnector
+
+            snowflake_credentials = SnowflakeCredentials(
+                account="account",
+                user="user",
+                password="password",
+            )
+            snowflake_connector = SnowflakeConnector(
+                database="database",
+                warehouse="warehouse",
+                schema="schema",
+                credentials=snowflake_credentials
+            )
+            with snowflake_connector.get_connection() as connection:
+                ...
+            ```
+        """
+        if self._connection is not None:
+            return self._connection
+
+        connect_params = {
+            "database": self.database,
+            "warehouse": self.warehouse,
+            "schema": self.schema_,
+        }
+        connection = self._credentials_get_client(**connect_kwargs, **connect_params)
+        self._connection = connection
+        self.logger.info("Started a new connection to Snowflake.")
+        return connection

```

Comment:
I like that idea more. I removed the relationship to prefect-snowflake and focused on keeping this a cursor class like `DictCursor`. It raises the question of "why is this in prefecto?", but it will be used alongside Prefect enough to be worth it as an extension.

```py
from snowflake.connector import connect
from prefecto.ext.snowflake import LogCursor

with connect(...).cursor(LogCursor) as cursor:
    cursor.execute(
        "SELECT * FROM table WHERE id = %(id)s AND secret = %(secret)s",
        params={"id": 123, "secret": "shhh"},
        obfuscate_params=["secret"],
        command_id="secret-selection",
    )
```

```
INFO - [secret-selection] Beginning command.
DEBUG - [secret-selection] Executing command:
SELECT * FROM table WHERE id = 123 AND secret = ****
INFO - [secret-selection] Command executed successfully.
```